table/tables/partition_test.go

```diff
@@ -470,6 +470,23 @@ func (ts *testSuite) TestRangePartitionUnderNoUnsignedSub(c *C) {
 	// MySQL will not support c1 value bigger than 9223372036854775817 in this case
 	_, err = tk.Exec(("insert into tu values (cast(18446744073709551615 as unsigned));"))
 	c.Assert(err, IsNil)
+	_, err = tk.Exec(("insert into tu values (cast(18446744073709551616 as unsigned));"))
+	c.Assert(err, NotNil)
+
+	// test `create table like`
+	tk.MustExec(`CREATE TABLE tu2 like tu;`)
```

Comment:
MySQL:
``` sql
mysql> SET @@sql_mode='NO_UNSIGNED_SUBTRACTION';
Query OK, 0 rows affected (0.00 sec)

mysql> CREATE TABLE tu2 like tu;
Query OK, 0 rows affected (0.02 sec)

mysql> SET @@sql_mode='';
Query OK, 0 rows affected (0.00 sec)

mysql> CREATE TABLE tu2 like tu;
ERROR 1563 (HY000): Partition constant is out of partition function domain
```

Sorry, I don't think you get my point. I want to ask you to test if `NO_UNSIGNED_SUBTRACTION
` affects `create table like`. To test it, you also need to set the `sql_mode` back.

Reply:
TiDB's `create table like` will [bypass partition table checks](https://github.com/pingcap/tidb/blob/master/ddl/ddl_api.go#L1593), and can successfully create the above table no matter which SQLMode it is in. This behaves different with MySQL. Should this be fixed in this pr?

Reply:
Also update this test in the new commit.